src/main/java/com/couchbase/lite/store/ForestDBStore.java

```diff
@@ -483,9 +484,27 @@ public RevisionList getAllRevisions(String docID, boolean onlyCurrent) {
                     int revFlags = (int) doc.getSelectedRevFlags();
                     if (((revFlags & C4RevisionFlags.kRevLeaf) != 0) == (leaf == 1 ? true : false) &&
                             RevisionInternal.generationFromRevID(revID) < generation) {
-                        revIDs.add(revID);
-                        if (outHaveBodies != null && !doc.hasRevisionBody())
+
+                        if (outHaveBodies != null && !doc.hasRevisionBody()) {
                             outHaveBodies.set(false);
+                            if (withBodiesOnly)
```

Comment:
I understand that this logic should be independent of whether outHaveBodies is null or not. The SQLiteStorage does that. Please double check.

Reply:
I guess @borrrden added this check to bypass the following block if `json` is null.

Reply:
Yes, I added a subsequent commit that addresses this https://github.com/couchbase/couchbase-lite-net/commit/76ef883aacf5eb00750a7771d35c1ab4645fd99b

Reply:
I wonder if `_removed=false` could be happening?

Reply:
I don't know if it could happen or not. I explicitly checked if the value is true or not in my recent change for the fix.

Reply:
A search of the sync_gateway repo shows that it only ever sets the value to a hard coded `true`

Reply:
I will add the check for _removed is true.